common/lib/helpers.py

```diff
@@ -40,6 +41,23 @@ def init_datasource(database, logger, queue, name):
     """
     pass
 
+# I was not completely happy with how either expand_url_shorteners or download_images was working
+# so I am trying regex. Plus we should have this as a helper function I think.
+# https://stackoverflow.com/questions/6038061/regular-expression-to-find-urls-within-a-string
+def extract_urls_from_string(text_with_urls, return_type='string'):
```

Comment:
We are using `ural` in some places, which has `urls_from_text()` and `urls_from_html()` functions we might use instead: https://github.com/medialab/ural?tab=readme-ov-file#urls_from_text

Reply:
I found issues with urls_from_text when I added this. I think it actually was failing on archive.org URLs since they have the http in the url twice. Perhaps it is more robust now.